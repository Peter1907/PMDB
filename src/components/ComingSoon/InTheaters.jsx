/* eslint-disable react-hooks/exhaustive-deps */
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getInTheaters, getStoredInTheaters } from '../Redux/in-theaters/in-theaters';
import { getStoredTrailer, getTrailer } from '../Redux/trailer/trailer';
import s from './InTheaters.module.css';

export default function InTheaters() {
  const playIcon = ('./assets/play.png');
  const starIcon = ('./assets/star.png');
  const saveIcon = ('./assets/add.png');
  const dispatch = useDispatch();

  const Data = useSelector((state) => state.inTheaters);
  const modData = Data.slice(1, 4);
  const DATA = localStorage.getItem('IN_THEATERS');

  const getData = () => (DATA ? dispatch(getStoredInTheaters()) : dispatch(getInTheaters()));

  useEffect(() => {
    getData();
  }, []);

  return (
    <div className={s.inTheaters}>
      <h1>In Theaters Now!</h1>
      <div className={s.block}>
        {(Data.length > 0)
          && (<div className={s.view}>
            <Thumbnail id={Data[0].id} />
            <div className={s.mainBlock}>
              <div className={s.imgContainer}>
                <img className={s.saveTag} src={saveIcon} alt="#" />
                <img src={Data[0].image} className={s.mainPoster} alt="poster" />
              </div>
              <div className={s.textContainer}>
                <h2>{Data[0].fullTitle}</h2>
                <p className={s.plot}>{`${(Data[0].plot).substring(0, 150)}...`}</p>
                <p className={s.mainPlayTrailer}>
                  <img src={playIcon} className={s.mainPlayButton} alt="play button" />
                  watch trailer
                </p>
              </div>
            </div>
          </div>)}
        <div className={s.next}>
          {modData.map((item) => (
            <div className={s.movie}>
              <img src={item.image} className={s.poster} alt="poster" />
              <div className={s.text}>
                <h3 className={s.title}>{item.title}</h3>
                <p><strong>Genre:</strong> {item.genres}</p>
                <p className={s.rating}>
                  <img src={starIcon} className={s.playButton} alt="start shaped icon" />
                  {item.imDbRating} ({item.imDbRatingCount})
                </p>
                <p className={s.playTrailer}>
                  <img src={playIcon} className={s.playButton} alt="play button" />
                  watch<br/> trailer
                </p>
              </div>
            </div>
          ))}
        </div>
      </div>
    </div>
  )
}

const Thumbnail = ({ id }) => {
  const dispatch = useDispatch();
  const storedTrailer = localStorage.getItem(id);
  const trailerInfo = useSelector((state) => state.trailer);

  const getData = () => {
    (storedTrailer) ? dispatch(getStoredTrailer(id)) : dispatch(getTrailer(id))
  };

  useEffect(() => {
    getData();
  }, []);

  return (
    <React.Fragment>
      <img src={trailerInfo.thumbnailUrl} alt="thumbnail" className={s.thumbnail} />
    </React.Fragment>
  );
};
