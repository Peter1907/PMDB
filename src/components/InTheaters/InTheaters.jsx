import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { getInTheaters, getStoredInTheaters, updateInTheaters } from '../Redux/in-theaters/in-theaters';
import Thumbnail from '../Thumbnail/Thumbnail';
import s from './InTheaters.module.css';

export default function InTheaters() {
  const playIcon = ('./assets/play.png');
  const starIcon = ('./assets/star.png');
  const saveIcon = ('./assets/add.png');
  const dispatch = useDispatch();

  const Data = useSelector((state) => state.inTheaters);
  const DATA = sessionStorage.getItem('IN_THEATERS');

  useEffect(() => {
    (DATA ? dispatch(getStoredInTheaters()) : dispatch(getInTheaters()));
  }, [DATA, dispatch]);

  const [count, setCount] = useState(7);
  const handleLoad = (i, event) => {
    dispatch(updateInTheaters(i));
    if (i === 7) event.target.parentNode.style.overflowY = 'scroll';
  };

  return (
    <div id="in-theaters" className={s.inTheaters}>
      <h1>In Theaters Now!</h1>
      {(Data.length > 0)
        && (<div className={s.block}>
          <div className={s.view}>
            <div className={s.thumbnailContainer}>
              <Thumbnail id={Data[0].id} />
              <div className={s.mainBlock}>
                <div className={s.imgContainer}>
                  <img className={s.saveTag} src={saveIcon} alt="#" />
                  <img src={Data[0].image} className={s.mainPoster} alt="poster" />
                </div>
                <div className={s.textContainer}>
                  <Link className={s.title} to={`/item-details/${Data[0].id}`}>
                    <h2>{Data[0].fullTitle}</h2>
                  </Link>
                  <p className={s.plot}>{`${(Data[0].plot).substring(0, 150)}...`}</p>
                  <p className={s.mainPlayTrailer}>
                    <img src={playIcon} className={s.mainPlayButton} alt="play button" />
                    watch trailer
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div className={s.next}>
            {Data[1].map((item, index) => (
              <div key={index} className={s.movie}>
                <img src={item.image} className={s.poster} alt="poster" />
                <div className={s.text}>
                  <Link to={`/item-details/${item.id}`}>
                    <h3 className={s.title} >{item.title}</h3>
                  </Link>
                  <p><strong>Genre:</strong> {item.genres}</p>
                  <p className={s.rating}>
                    <img src={starIcon} className={s.playButton} alt="start shaped icon" />
                    {item.imDbRating} ({item.imDbRatingCount})
                  </p>
                  <p className={s.playTrailer}>
                    <img src={playIcon} className={s.playButton} alt="play button" />
                    watch<br /> trailer
                  </p>
                </div>
              </div>
            ))}
            <button
              className={s.loadBtn}
              type="button"
              onClick={(event) => {
                handleLoad(count, event);
                setCount(count + 3);
              }}
            >
              Load more
            </button>
          </div>
        </div>)}
    </div>
  )
}
