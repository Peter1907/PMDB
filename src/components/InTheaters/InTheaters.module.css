.inTheaters {
  padding: 5rem 2rem 3rem;
  background: linear-gradient(#222831, #222831, #393e46);
  color: #eee;
  margin: 0;
  font-family: 'Cairo', sans-serif;
}

.inTheaters h1 {
  color: #ffd369;
  font-family: 'Barlow Semi Condensed', sans-serif;
  margin: 0 0 3.5rem;
  text-align: center;
  font-size: 72px;
}

.block {
  margin: auto;
  max-width: 75rem;
  display: flex;
  gap: 1rem;
}

.view {
  width: 68.5%;
  position: relative;
  background-color: black;
}

.thumbnailContainer {
  min-height: 28.7rem;
  background-color: black;
}

.mainBlock {
  position: absolute;
  bottom: 0;
  left: 7%;
  width: 90%;
  background: linear-gradient(rgba(0, 0, 0, 0.151), black);
  backdrop-filter: blur(1px);
  display: flex;
  gap: 3rem;
}

.mainImgContainer {
  position: relative;
  width: 100%;
  max-width: 10rem;
  overflow: hidden;
}

.saveTag {
  position: absolute;
  top: 0;
  left: 7%;
  width: 40px;
  opacity: 0.3;
  cursor: pointer;
}

.saveTag:hover {
  opacity: 0.8;
  box-shadow: inset 0 0 0 80px #ffd369;
}

.textContainer h2 {
  font-size: 42px;
  font-family: 'Barlow Semi Condensed', sans-serif;
  font-weight: 600;
  color: #ffd369;
  margin: 0;
}

.plot {
  font-size: 18px;
  margin: 0 3%;
}

.mainPlayTrailer {
  display: flex;
  font-size: 32px;
  gap: 1rem;
  align-items: center;
  cursor: pointer;
  margin: 0;
  background-color: rgba(0, 0, 0, 0.247);
  width: 100%;
  padding: 10px;
}

.mainPlayTrailer:hover {
  color: #ffd369;
}

.mainPlayButton {
  width: 65px;
}

.next {
  width: 30%;
  max-width: 30rem;
  display: flex;
  flex-direction: column;
  gap: 5px;
  max-height: 40.2rem;
}

.text {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
}

.text p {
  margin: 0;
}

.rating {
  display: flex;
  gap: 0.5rem;
  align-items: center;
}

.rating img {
  width: 20px;
}

.playTrailer {
  display: flex;
  align-items: center;
  margin: 0;
  line-height: 1.2rem;
  gap: 0.5rem;
  cursor: pointer;
}

.playTrailer:hover {
  color: #ffd369;
}

.playButton {
  width: 38px;
}

.movie {
  display: flex;
  align-items: center;
  gap: 1rem;
  background: linear-gradient(rgba(0, 0, 0, 0.555), #222831a4);
  padding: 10px;
}

.imgContainer {
  position: relative;
  width: 100%;
  max-width: 7rem;
  height: 10rem;
  overflow: hidden;
}

.title {
  font-weight: 700;
  margin: 0;
  color: #ffd369;
}

.loadBtn {
  background-color: rgba(0, 0, 0, 0.466);
  border: none;
  font-size: 16px;
  height: 2.7rem;
  font-weight: 700;
  color: #eee;
  cursor: pointer;
  flex-shrink: 0;
}

.loadBtn:hover {
  background-color: rgba(0, 0, 0, 0.658);
  color: #ffd369;
}

.loadBtn:active {
  transform: scale(0.98);
}

@media screen and (max-width: 1024px) {
  .block {
    flex-direction: column;
  }

  .view {
    width: 100%;
    overflow: hidden;
    aspect-ratio: 14 / 9;
  }

  .next {
    width: 70%;
    margin: auto;
  }
}

@media screen and (max-width: 768px) {
  .inTheaters {
    padding: 5rem 5% 3rem;
  }

  .inTheaters h1 {
    text-align: center;
    font-size: 60px;
  }

  .block {
    flex-direction: column;
  }

  .view {
    width: 100%;
    min-height: 75rem;
  }

  .mainBlock {
    position: static;
    margin: auto auto 20rem;
    display: flex;
    background: transparent;
    flex-direction: column;
    align-items: flex-start;
    gap: 0.5rem;
    padding: 7rem 0 0;
  }

  .mainImgContainer {
    max-width: 350px;
    margin: auto;
  }

  .textContainer h2 {
    font-size: 34px;
    position: absolute;
    top: 1.5rem;
    left: 5%;
  }

  .mainPlayTrailer {
    font-size: 26px;
    position: absolute;
    bottom: -25rem;
  }

  .mainPlayButton {
    width: 50px;
  }

  .next {
    width: 100%;
  }
}

@media screen and (max-width: 436px) {
  .inTheaters {
    padding: 5rem 0.5rem 3rem;
  }

  .inTheaters h1 {
    font-size: 46px;
  }
}
