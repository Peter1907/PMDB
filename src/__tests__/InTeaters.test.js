import renderer from 'react-test-renderer';
import InTheaters from '../_mocks_/InTheaters';

it('renders correctly', () => {
  const tree = renderer
    .create(<InTheaters />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
