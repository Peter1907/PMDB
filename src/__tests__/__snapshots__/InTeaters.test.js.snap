// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="inTheaters"
  id="in-theaters"
>
  <h1>
    In Theaters Now!
  </h1>
  <div
    className="block"
  >
    <div
      className="view"
    >
      <div
        className="thumbnailContainer"
      >
        <img
          alt="thumbnail"
          className="thumbnail"
          src="https://m.media-amazon.com/images/M/MV5BZjc3YjUxMzYtNTVlYy00YjhkLTg1YzctNjljYmY4OWRkMDVkXkEyXkFqcGdeQWRpZWdtb25n._V1_.jpg"
        />
        <div
          className="mainBlock"
        >
          <div
            className="imgContainer"
          >
            <img
              alt="#"
              className="saveTag"
              src="./assets/add.png"
            />
            <img
              alt="poster"
              className="mainPoster"
              src="https://m.media-amazon.com/images/M/MV5BZTg1Y2Q3MzctMDlkOS00OGE1LWE4MjUtNmJjNDNkZmM2YmVkXkEyXkFqcGdeQXVyMjY5ODI4NDk@._V1_Ratio0.6699_AL_.jpg"
            />
          </div>
          <div
            className="textContainer"
          >
            <a
              className="title"
              href="/item-details/tt10665342"
              onClick={[Function]}
            >
              <h2>
                Halloween Ends (2022)
              </h2>
            </a>
            <p
              className="plot"
            >
              The saga of Michael Myers and Laurie Strode comes to a spine-chilling climax in this final installment of this trilogy....
            </p>
            <p
              className="mainPlayTrailer"
            >
              <img
                alt="play button"
                className="mainPlayButton"
                src="./assets/play.png"
              />
              watch trailer
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="next"
    >
      <div
        className="movie"
      >
        <img
          alt="poster"
          className="poster"
          src="https://m.media-amazon.com/images/M/MV5BZjE2ZWIwMWEtNGFlMy00ZjYzLWEzOWEtYzQ0MDAwZDRhYzNjXkEyXkFqcGdeQXVyMTUzMTg2ODkz._V1_Ratio0.6699_AL_.jpg"
        />
        <div
          className="text"
        >
          <a
            href="/item-details/tt15474916"
            onClick={[Function]}
          >
            <h3
              className="title"
            >
              Smile (V) (2022)
            </h3>
          </a>
          <p>
            <strong>
              Genre:
            </strong>
             
            Horror
          </p>
          <p
            className="rating"
          >
            <img
              alt="start shaped icon"
              className="playButton"
              src="./assets/star.png"
            />
            6.9
             (
            25377
            )
          </p>
          <p
            className="playTrailer"
          >
            <img
              alt="play button"
              className="playButton"
              src="./assets/play.png"
            />
            watch
            <br />
             trailer
          </p>
        </div>
      </div>
      <div
        className="movie"
      >
        <img
          alt="poster"
          className="poster"
          src="https://m.media-amazon.com/images/M/MV5BNDQwNzE0ZTItYmZjMC00NjI3LWFlNzctNTExZDY2NWE0Zjc0XkEyXkFqcGdeQXVyMTA3MDk2NDg2._V1_Ratio0.6699_AL_.jpg"
        />
        <div
          className="text"
        >
          <a
            href="/item-details/tt10304142"
            onClick={[Function]}
          >
            <h3
              className="title"
            >
              Amsterdam
            </h3>
          </a>
          <p>
            <strong>
              Genre:
            </strong>
             
            Comedy, Drama, History
          </p>
          <p
            className="rating"
          >
            <img
              alt="start shaped icon"
              className="playButton"
              src="./assets/star.png"
            />
            6.2
             (
            9671
            )
          </p>
          <p
            className="playTrailer"
          >
            <img
              alt="play button"
              className="playButton"
              src="./assets/play.png"
            />
            watch
            <br />
             trailer
          </p>
        </div>
      </div>
      <div
        className="movie"
      >
        <img
          alt="poster"
          className="poster"
          src="https://m.media-amazon.com/images/M/MV5BMDU2ZmM2OTYtNzIxYy00NjM5LTliNGQtN2JmOWQzYTBmZWUzXkEyXkFqcGdeQXVyMTkxNjUyNQ@@._V1_Ratio0.6699_AL_.jpg"
        />
        <div
          className="text"
        >
          <a
            href="/item-details/tt12593682"
            onClick={[Function]}
          >
            <h3
              className="title"
            >
              Bullet Train
            </h3>
          </a>
          <p>
            <strong>
              Genre:
            </strong>
             
            Action, Comedy, Thriller
          </p>
          <p
            className="rating"
          >
            <img
              alt="start shaped icon"
              className="playButton"
              src="./assets/star.png"
            />
            7.4
             (
            140434
            )
          </p>
          <p
            className="playTrailer"
          >
            <img
              alt="play button"
              className="playButton"
              src="./assets/play.png"
            />
            watch
            <br />
             trailer
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
